added authfetch utility

danielbrog opened the pull request:
fixed how server sends error messaging
added folder/file utilities/auth for authFetch call
replaced all fetch calls with authFetch in client

danielbrog committed:
added authfetch utility

fixed how server sends error messaging
added folder/file utilities/auth for authFetch call
replaced all fetch calls with authFetch in client

aeciorc commented on server/routes/user.js:
```diff
@@ -86,6 +82,7 @@ router.get("/user/:id", auth, async (req, res) => {
     await user.save();
     res.status(200).send(user);
   } catch (e) {
+    res.statusMessage = e;
```
Make sure that e is string-friendly. Maybe you will need to use e.message instead

danielbrog committed:
Fixed error status messages

fixed the backend status messages sent to the front end to make sure strings are sent

danielbrog committed:
Merge branch 'dev' into FE-utility-with-auth
